libs/api-mocks/msw/util.ts

```diff
@@ -292,24 +294,45 @@ export function currentUser(req: RestRequest): Json<User> {
 }
 
 /**
- * Look for fleet roles in roles for the user as well as for groups the user is
- * in. Testable core of `requireFleetViewer`.
+ * Given a role A, get a list of the roles (including A) that confer *at least*
+ * the powers of A.
  */
-export function userIsFleetViewer(user: Json<User>): boolean {
+// could implement with `takeUntil(allRoles, r => r === role)`, but that is so
+// much harder to understand
+const roleOrStronger: Record<RoleKey, RoleKey[]> = {
+  viewer: ['viewer', 'collaborator', 'admin'],
+  collaborator: ['collaborator', 'admin'],
+  admin: ['admin'],
+}
+
+/**
+ * Determine whether a user has a role at least as strong as the specified role
+ * on the specified resource.
+ */
+export function userHasRole(
+  user: Json<User>,
+  resourceType: DbRoleAssignmentResourceType,
+  resourceId: string,
+  role: RoleKey
+): boolean {
   const userGroupIds = db.groupMemberships
     .filter((gm) => gm.userId === user.id)
     .map((gm) => db.userGroups.find((g) => g.id === gm.groupId))
     .filter(isTruthy)
     .map((g) => g.id)
 
-  // don't need to filter by role because any role is at least viewer
-  const actorsWithFleetRole = db.roleAssignments
-    .filter((ra) => ra.resource_type === 'fleet')
+  /** All actors with *at least* the specified role on the resource */
+  const actorsWithRole = db.roleAssignments
+    .filter(
+      (ra) =>
+        ra.resource_type === resourceType &&
+        ra.resource_id === resourceId &&
+        roleOrStronger[role].includes(ra.role_name)
+    )
```

Comment:
Honestly this was very easy. Not sure why I didn't do this before?

Well.... this does not account for parent-child relationships, like if you have collab on silo you also have it on all projects in silo. I need to either account for that or add a comment about the limitation.